src/test/scala/edu/ie3/simona/config/ConfigFailFastSpec.scala

```diff
@@ -870,7 +871,76 @@ class ConfigFailFastSpec extends UnitSpec with ConfigTestData {
             )
           }.getMessage shouldBe "The weather data scheme 'this won't work' is not supported. Supported schemes:\n\ticon\n\tcosmo"
         }
+      }
+
+      "checking the transformer control groups" should {
+        val checkTransformerControl =
+          PrivateMethod[Unit](Symbol("checkTransformerControl"))
+
+        "throw an exception, if the measurements are empty" in {
+          val dut = TransformerControlGroup(
+            List.empty[String],
+            List("a16cf7ca-8bbf-46e1-a74e-ffa6513c89a8"),
+            1.02,
+            0.98
+          )
+
+          intercept[InvalidConfigParameterException] {
+            ConfigFailFast invokePrivate checkTransformerControl(dut)
+          }.getMessage shouldBe "A transformer control group cannot have no measurements assigned."
+        }
+
+        "throw an exception, if the transformers are empty" in {
+          val dut = TransformerControlGroup(
+            List("6888c53a-7629-4563-ac8e-840f80b03106"),
+            List.empty[String],
+            1.02,
+            0.98
+          )
 
+          intercept[InvalidConfigParameterException] {
+            ConfigFailFast invokePrivate checkTransformerControl(dut)
+          }.getMessage shouldBe "A transformer control group cannot have no transformers assigned."
+        }
+
```

Comment:
This test has its list of strings swapped, I guess. Not a major issue because it should fail if either of the strings are empty.